### PR TITLE
SDI-459 prisoner-offender-search.prisoner.alerts-updated event

### PR DESCRIPTION
**Pull request by andymarke**
* raise new event any time the list of alerts change

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ class PrisonerMovementsEventService(
   private val domainEventEmitter: HmppsDomainEventEmitter,
   private val telemetryClient: TelemetryClient,
 ) {
-  fun generateAnyMovementEvents(
+  fun generateAnyEvents(
     previousPrisonerSnapshot: Prisoner?,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:32]: minor name change to so all services simply have a `generateAnyEvents` method, the type is implied by the service name

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/AlertsUpdatedEventService.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@Service
+class AlertsUpdatedEventService(
+  private val domainEventEmitter: HmppsDomainEventEmitter,
+) {
+  fun generateAnyEvents(
+    previousPrisonerSnapshot: Prisoner?,
+    prisoner: Prisoner
+  ) {
+    val previousAlertList = previousPrisonerSnapshot?.alerts?.map { it.alertCode }?.toSet() ?: emptySet()
+    val alertList = prisoner.alerts?.map { it.alertCode }?.toSet() ?: emptySet()
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/AlertsUpdatedEventService.kt:16]: trivial point which looks slightly odd - your list variables here are actually sets

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/AlertsUpdatedEventService.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/AlertsUpdatedEventService.kt:16]: Yeah, I'll rename in next pr

**Pull request merged**